sidecar/proxy/envoy/manager.go

```diff
@@ -196,59 +198,47 @@ func (m *manager) generateConfig(rules []api.Rule, instances []api.ServiceInstan
 
 	format := fmt.Sprintf(envoyLogFormat, buildSourceName(inst.ServiceName, inst.Tags), traceKey, traceVal)
 
+	listeners := BuildListeners(m.listenerPort, filters, format, m.loggingDir+accessLog, m.tcpProxyConfigs)
+
+	staticClusters := []Cluster{
+		{
+			Name:             "rds",
+			Type:             "strict_dns",
+			ConnectTimeoutMs: 1000,
+			LbType:           "round_robin",
+			Hosts: []Host{
+				{
+					URL: fmt.Sprintf("tcp://127.0.0.1:%v", m.sdsPort),
+				},
+			},
+			MaxRequestsPerConnection: 1,
+		},
+	}
+
+	for _, proxy := range m.tcpProxyConfigs {
+		cluster := Cluster{
+			Name:             proxy.Service,
+			ServiceName:      proxy.Service,
```

Comment:
TCP filters have a statically defined cluster name.  Envoy rejects the config if there is not a statically defined cluster in cluster_manager.  CDS is not compatible with TCP at this point